### ISSUE TITLE
[R-Forge #6092] perTradeStats Net.Trading.PL Error

### ISSUE THREAD

**Issue by joshuaulrich** [braverock/blotter#23]
Submitted by: Xie Wenfei
Assigned to: Nobody
[R-Forge link](http://r-forge.r-project.org/tracker/index.php?func=detail&aid=6092&group_id=316&atid=1269)

I am using latest versions of all libraries. 
I met a problem. I should be a bug. 
Net.Trading.PL should be calcuated by (2021.285 - 2024.951) \* 8468 = -31043.688. 
But in perTradeStats Net.Trading.PL result 112488.912

> tail(book$holyGrail1$SINAPLATEINFO)
>            Order.Qty Order.Price Order.Type Order.Side Order.Threshold Order.Status Order.StatusTime      Prefer  Order.Set Txn.Fees
> 2014-05-21 '8468'    '2024.951'  'market'   'long'     NA              'closed'     '2014-05-22 00:00:00' 'Close' NA        '0'  
> 2014-05-22 'all'     '2021.285'  'market'   'long'     NA              'closed'     '2014-05-23 00:00:00' ''      NA        '0'     
> 
> tail(perTradeStats(q.strategy, includeOpenTrade = FALSE))
>         Start        End Init.Pos Max.Pos Num.Txns Max.Notional.Cost Net.Trading.PL        MAE         MFE Pct.Net.Trading.PL
> 41 2014-05-22 2014-05-23     8468    8468        2          17116241     112488.912       0.00  112488.912       0.0065720569

My R code is attached:

  add.signal(q.strategy, name = 'sigCrossover', arguments = list(columns = c('Close',
                                                                             'TrendMA'), relationship = 'gte'), label = 'Cl.gt.TrendMA')
  add.signal(q.strategy, name = 'sigCrossover', arguments = list(columns = c('Close',
                                                                             'TrendMA'), relationship = 'lt'), label = 'Cl.lt.TrendMA')

  add.signal(q.strategy,name='sigThreshold',
             arguments = list(column='WinLosePredict',
                              relationship='eq',
                              threshold=1,
                              cross=FALSE),
             label='WinLosePredict.LongWin')

  add.rule(q.strategy, name = 'ruleSignal', arguments = list(sigcol = 'Cl.lt.TrendMA',
                                                             sigval = TRUE, orderqty = 'all', ordertype = 'market', orderside = 'long', osFUN = 'osMaxPos',
                                                             pricemethod = 'market', replace=TRUE), type = 'exit', path.dep = TRUE, label = 'ExitLONG')

  add.rule(q.strategy, name = 'ruleSignal', arguments = list(sigcol = 'WinLosePredict.LongWin',
                                                             sigval = TRUE, orderqty = pos, ordertype = 'market', prefer='Close',orderside = 'long', osFUN = 'osMaxPos',
                                                             pricemethod = 'market', replace=FALSE), type = 'enter', path.dep = TRUE, label = 'EnterLONG')

  out <- applyStrategy(strategy = q.strategy, portfolios = q.strategy)

  updatePortf(q.strategy)

  updateAcct(q.strategy)

  updateEndEq(q.strategy)

Followups:

Date: 2015-05-21 23:48
Sender: Xie Wenfei
Dear Brian,   Thanks so much for your quick response. I really appreciate your brilliant work in these R projects.    Well, I will clarify it with demo example. I commented 'long' add.rule lines and only left 'short' add.rule lines in luxor.1.strategy.basic. You can run luxor.1.strategy.basic as attached. From those diagrams generated, we can find that Issue 1:In perTradeStatsNet.Trading.PL   -390    -250     170    -650      40I suppose we can calculate perTradeStats from getOrderBook. How Net.Trading.PL is calculated from getOrderBook? Could you explain how -250 is calculated for example?Issue 2: In the diagrams of both 'chart.Posn(portfolio.st, GBPUSD)' and 'charts.PerformanceSummary(ret1$total, geometric=FALSE, wealth.index=TRUE)', we can see that Cumulative Return and Drawback Daily return is so poor.  I doubt the tradeStats calculation of short trades. Finally, we really appreciate your team for these excellent projects to explore market secret. Looking forward to your reply. 

Date: 2015-05-21 09:13
Sender: Brian Peterson
Your example is not reproducible.At the very least, pleas attach the data which generates what you believe to be a problem.Better yet, produce an actual minimal reproducible example.There are a number of ways to do this...Probably the simplest aside from attaching the data used to generate your test would be to call extractTxns on the portfolio which you believe contains an error, and extract that along with an explanation of where you think the error occurs.Even better would be a small crafted set of trades which replicates what you believe to be the error.Otherwise, I can't evaluate your hand waving, and can't help you.

**Comment by braverock** [comment]
closing either not reproducible or as likely fixed by changes made to #46 and #47

**Comment by joshuaulrich** [comment]
This was likely never an issue. The order book is:

```r
           Order.Qty Order.Price Order.Type Order.Side Order.Threshold Order.Status      Order.StatusTime  Prefer Order.Set Txn.Fees
2014-05-21    "8468"  "2024.951"   "market"     "long"              NA     "closed" "2014-05-22 00:00:00" "Close"        NA      "0" 
2014-05-22     "all"  "2021.285"   "market"     "long"              NA     "closed" "2014-05-23 00:00:00"      ""        NA      "0" 
```
And the OP expected the Net P&L of: `(2021.285 - 2024.951) * 8468 = -31043.688`.  Notice that they used the order prices, but `Order.Type = "market"` and `Order.StatusTime = [t+1]`.

Since the two orders were placed on back-to-back observations, we know the entry was filled at 2021.285.  And a Net P&L of 112,488.912 implies an exit price of: `112488.912 / 8468 + 2024.951 = 2038.235`, which is not an unreasonable expectation for the price on 2014-05-23.

Removing "bug" and adding "invalid" label.
